### ISSUE TITLE
[Feature]: Support Vue SSG (prerendering)

### ISSUE THREAD

**Issue by MuTsunTsai** [web-infra-dev/rsbuild#2894]
### What problem does this feature solve?

Some of my projects are using Vue SSG (see [[1](https://vuejs.org/guide/extras/ways-of-using-vue#jamstack-ssg)] [[2](https://vuejs.org/guide/scaling-up/ssr#ssr-vs-ssg)]) to improve loading performance, and I'm hoping that Rsbuild vue plugin can add support to this feature so that I may migrate those projects to Rsbuild.

Currently for webpack there is [Prerenderer](https://github.com/Tofandel/prerenderer) that does similar things, but it does not support Rspack.

### What does the proposed API look like?

Ideally it will be a one-option setup:
```js
pluginVue({
  prerendering: true,
});
```

**Comment by chenjiahan** [comment]
The SSG feature should be provided by frameworks such as Rspress or Nuxt. 

`@rsbuild/plugin-vue` will focus on providing primary support for Vue, like SFC compilation.

**Comment by MuTsunTsai** [comment]
@chenjiahan You have a point, but my projects are not using those frameworks though. Perhaps forking @prerenderer/webpack-plugin to make it work on Rspack?

**Comment by chenjiahan** [comment]
Are you using `@prerenderer/webpack-plugin` at present?

**Comment by MuTsunTsai** [comment]
@chenjiahan Nope. Those old projects of mine are not even using webpack in fact (but mainly with Gulp + esbuild, if that matters). I noticed that @prerenderer/webpack-plugin doesn't work as I started a new project with Rsbuild experimenting the features I need (so that I may migrate with confidence once all features are supported).
